File: problems/algorithms/array-matrix-operations/js/response/matrix-operations.js
function addingMatrix(a, b) {
  let sum = [];
  for (var row = 0; row < a.length; row++) {
    let temp = [];
    for (var col = 0; col < a[0].length; col++) {
      temp[col] = a[row][col] + b[row][col];
    }
    sum.push(temp);
  }
  return sum;
}

function multiplyingMatrix(a, b) {
  let result = [];
  for (var x = 0; x < a.length; x++) {
    let temp = [];
    for (var y = 0; y < a.length; y++) {
      let sum = 0;
      for (var z = 0; z < b.length; z++) {
        sum += a[x][z] * b[z][y];
      }
      temp.push(sum);
    }
    result.push(temp);
  }
  return result;
}

export { addingMatrix, multiplyingMatrix };

File: problems/algorithms/array-matrix-operations/ruby/response/src/matrix_operations.rb
def addingMatrix(a, b)
  sum = []
  a.each_with_index do |row,indexRow|
    temp = []
    row.each_with_index do |col,indexCol|
      temp[indexCol] = a[indexRow][indexCol] + b[indexRow][indexCol]
    end
    sum.push(temp)
  end 
return sum
end

def multiplyingMatrix(a, b)
  result = []
  (0..a.length-1).step(1) do |x|
    temp = []
    (0..a.length-1).step(1) do |y|
      sum=0
      (0..b.length-1).step(1) do |z|
       sum = sum + (a[x][z] * b[z][y])
      end
      temp.push(sum)
    end
    result.push(temp)
end
return result
end

